### DISCUSSION TITLE
How to handle errors inside a runInAction?

### DISCUSSION THREAD

**Discussion by SamuelPinho** [mobxjs/mobx#3707] [Q&A] [answered]
Imagine that the following code inside the `runInAction`, does throw an error:

```tsx
runInAction(() => {
  // items variable for some reason is a "string" so it will throw an error
  // when trying to access items.length
  if (this.items.length === 0 && totalItemCount > 0) {
    this.items = new Array(totalItemCount);
  }
});
```

Whenever an error occurs inside a `runInAction`, I can't catch it through a `try/catch` block. I read (on MobX documentation) that this might be something done on purpose, but I wonder how I should handle this problem 🤔 

_I'm using React, so I could have an ErrorBoundary, but are there any other solutions to that?_

**Comment by mweststrate** [comment, marked as answer]
RunInAction will throw normally, so you can catch it. ErrorBoundaries only
catch errors from rendering, not from event handling iirc.

**Reply by SamuelPinho** [reply]
You're right. Thanks for that. I checked my code again and it wasn't throwing because of an issue where trying to cause the error wasn't triggering the `runInAction`.

Thanks again

**Comment by mweststrate** [comment]
Note that items, being a string, does have a length property, so it won't
throw. And if it hadn't, it would return undefined, still not throw.
